### PR TITLE
dev/core#1486 Remove Foreign Keys from ACL Cache tables and replace o…

### PR DESCRIPTION
**Pull request by seamuslee001**
…ne with an index on contact_id

Overview
----------------------------------------
This removes the Foreign Keys from the ACL cache tables as these are likely to only hinder performance than help

Before
----------------------------------------
FKs on ACL cache tables

After
----------------------------------------
No FKs on ACL cache tables

ping @eileenmcnaughton @kcristiano

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by kcristiano** [comment]
@seamuslee001 We have a site running this in production with no issues.   Looks merge ready to me.

**Comment by eileenmcnaughton** [comment]
Do we still have an index on acl_id?

**Comment by kcristiano** [comment]
you are correct we do.   @seamuslee001 perhaps we don't drop `FK_civicrm_acl_cache_acl_id`

**Comment by seamuslee001** [comment]
@eileenmcnaughton @kcristiano I have now updated this as per Kevin's comment

**Comment by eileenmcnaughton** [comment]
Did my triage tinkering trigger you to do that ?

**Comment by seamuslee001** [comment]
6/1 0.5 of a dozen

**Comment by eileenmcnaughton** [comment]
I just thought of something else - on upgrade we remove the fks but don't add the indices?

**Comment by seamuslee001** [comment]
@eileenmcnaughton isn't there that system api job to build the indices I thought that was the way sites were to get the new indexes or has things changed?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 yes - but sites have to run it & they no longer get status checks to do so. In most cases adding indices is very optional so no issue if they don't run it. In this case though we are actively removing something index-like so it would feel like a regression.

I suspect if you go round client sites & run the index script you will find a few are missing recommended indexes. Having said that, on wmf we've started removing recommended indices - most notably on contribution_status_id based on query testing so if we ran it we'd get indices we really don't want.

Short version = in this case it's probably better to include the index add

**Comment by seamuslee001** [comment]
@eileenmcnaughton have added the code to add the index on upgrade now

**Comment by eileenmcnaughton** [comment]
@seamuslee001 OK - I think that's fine - @agh1 can you give this a bit of prominence in the release notes in case any DBs play weirdly with the index & FK changes

File: CRM/Core/BAO/SchemaHandler.php [hunk 1]
```diff
@@ -365,9 +365,9 @@ public static function changeUniqueToIndex($tableName, $dropUnique = TRUE) {
    *
    * @param $tables
    *   Tables to create index for in the format:
-   *     array('civicrm_entity_table' => 'entity_id')
```
Comment (seamuslee001) [inline CRM/Core/BAO/SchemaHandler.php:368]: Note that this is just documentation but it also reflects the fact that the function expects all fields to be in an array even if just one now
